newIDE/app/src/EventsFunctionsList/index.js

```diff
@@ -37,7 +37,17 @@ export type EventsFunctionCreationParameters = {|
 |};
 
 const getEventsFunctionName = (eventsFunction: gdEventsFunction) =>
-  eventsFunction.getName() + (eventsFunction.isPrivate() ? ' (private)' : '');
+  eventsFunction.isPrivate() ? (
+    <React.Fragment>
+      <VisibilityOffIcon
+        fontSize="small"
+        style={{ marginRight: 5, verticalAlign: 'bottom' }}
+      />
+      {eventsFunction.getName()}
+    </React.Fragment>
```

Comment:
Nitpick: you do not need to write out `React.Fragment`, instead you can just use empty brackets:
```suggestion
    <>
      <VisibilityOffIcon
        fontSize="small"
        style={{ marginRight: 5, verticalAlign: 'bottom' }}
      />
      {eventsFunction.getName()}
    </>
```

Reply:
To resolve the title issue in this case, you can change the title props to the div in `ItemRow` to be:

```
title={typeof itemName === 'string' ? itemName : undefined}
```

And you can set your own title here, putting the `eventsFunction.getName()` in a span.

Reply:
You would need to change the type of the props itemName to be `string | React.Element` or something like this to be correct